galley/pkg/config/analysis/analyzers/virtualservice/gateways.go

```diff
@@ -59,7 +61,19 @@ func (s *GatewayAnalyzer) analyzeVirtualService(r *resource.Entry, c analysis.Co
 			continue
 		}
 
-		if !c.Exists(metadata.IstioNetworkingV1Alpha3Gateways, resource.NewName(ns, gwName)) {
+		gwRName, err := resource.NewFullName(gwName) // Incorrect name should be catched in gw validations
+		if err != nil {
+			c.Report(metadata.IstioNetworkingV1Alpha3Virtualservices,
+				msg.NewParseError(r, fmt.Sprintf("Invalid gateway name %s: %s", gwName, err.Error())))
+
+		}
```

Comment:
Would our schema-based validation (applied both via the validation webhook, and also in a separate analyzer) have caught this? Does that make this check redundant?

This seems like the kind of thing the webhook *should* be covering, if it isn't already.

Reply:
I tested my variant with the gateway `istio-system/` and with `/helloworld-gateway` but did not check in those tests.

At least for my PR, schema validation reports this problem.  However I did not like how schema validation reported it, so I kept in the redundant message.

```
Error [IST0101] (VirtualService helloworld.default) Referenced gateway not found: "istio-system/"
Error [IST0106] (VirtualService helloworld.default) Schema validation error: config namespace and gateway name cannot be empty
Error [IST0106] (VirtualService helloworld.default) Schema validation error: invalid value for gateway name: ""
Error: Analyzer found issues.
```